### PR TITLE
Power Distribution Device ID -> core

### PR DESCRIPTION
**Pull request by drorta**

### CODE REVIEW

File: src/main/java/frc/utils/battery/BatteryConstants.java [hunk 1]
```diff
@@ -8,7 +8,9 @@ class BatteryConstants {
 	protected static final String LOG_PATH = "Battery/";
 	protected static final String ALERT_LOG_PATH = LogPaths.ALERT_LOG_PATH + LOG_PATH;
 
-	protected static final int POWER_DISTRIBUTION_CAN_ID = 20;
-	protected static final PowerDistribution.ModuleType POWER_DISTRIBUTION_TYPE = PowerDistribution.ModuleType.kRev;
+	protected static final PowerDistributionDeviceID POWER_DISTRIBUTION_DEVICE_ID = new PowerDistributionDeviceID(
+		20,
+		PowerDistribution.ModuleType.kRev
+	);
```
Comment (YHerm) [inline src/main/java/frc/utils/battery/BatteryConstants.java:14]: put in class ID's

File: src/main/java/frc/utils/battery/PowerDistributionDeviceID.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+		this.id = id;
+		return this;
+	}
+
+	public PowerDistributionDeviceID withType(PowerDistribution.ModuleType type) {
+		this.type = type;
+		return this;
+	}
+
+}
```
Comment (YHerm) [inline src/main/java/frc/utils/battery/PowerDistributionDeviceID.java:33]: consider making it a record

**Review by YHerm — CHANGES_REQUESTED** [review]

File: src/main/java/frc/utils/battery/BatteryConstants.java (hunk 1 above)
Comment (drorta) [reply src/main/java/frc/utils/battery/BatteryConstants.java:14]: k

File: src/main/java/frc/utils/battery/PowerDistributionDeviceID.java (hunk 2 above)
Comment (YHerm) [reply src/main/java/frc/utils/battery/PowerDistributionDeviceID.java:33]: kind of class

**Review by GilStein1 — DISMISSED** [review]
very good👌

File: src/main/java/frc/utils/battery/BatteryConstants.java [hunk 3]
```diff
@@ -8,7 +8,6 @@ class BatteryConstants {
 	protected static final String LOG_PATH = "Battery/";
 	protected static final String ALERT_LOG_PATH = LogPaths.ALERT_LOG_PATH + LOG_PATH;
 
-	protected static final int POWER_DISTRIBUTION_CAN_ID = 20;
-	protected static final PowerDistribution.ModuleType POWER_DISTRIBUTION_TYPE = PowerDistribution.ModuleType.kRev;
+	protected static final PowerDistribution.ModuleType DEFAULT_MODULE_TYPE = PowerDistribution.ModuleType.kRev;
```
Comment (YHerm) [inline src/main/java/frc/utils/battery/BatteryConstants.java:11]: no need default

File: src/main/java/frc/utils/battery/PowerDistributionDeviceID.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package frc.utils.battery;
+
+import edu.wpi.first.wpilibj.PowerDistribution;
+
+public record PowerDistributionDeviceID(int id, PowerDistribution.ModuleType type) {
```
Comment (YHerm) [inline src/main/java/frc/utils/battery/PowerDistributionDeviceID.java:5]:
```suggestion
public record PowerDistributionDeviceID(int ID, PowerDistribution.ModuleType type) {
```

File: src/main/java/frc/utils/battery/PowerDistributionDeviceID.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+	public PowerDistributionDeviceID(int id) {
+		this(id, BatteryConstants.DEFAULT_MODULE_TYPE);
+	}
```
Comment (YHerm) [inline src/main/java/frc/utils/battery/PowerDistributionDeviceID.java:9]: remove

File: src/main/java/frc/utils/battery/BatteryUtils.java [hunk 6]
```diff
@@ -5,17 +5,16 @@
 import edu.wpi.first.wpilibj2.command.Command;
 import org.littletonrobotics.junction.Logger;
 
-import static frc.utils.battery.BatteryConstants.POWER_DISTRIBUTION_CAN_ID;
-import static frc.utils.battery.BatteryConstants.POWER_DISTRIBUTION_TYPE;
+import static frc.robot.constants.IDs.*;
```
Comment (YHerm) [inline src/main/java/frc/utils/battery/BatteryUtils.java:8]: dont use *

**Review by YHerm — APPROVED** [review]

**Review by YHerm — APPROVED** [review]

**Review by YoniKiriaty — DISMISSED** [review]

**Review by YHerm — APPROVED** [review]

**Review by Sh-Pe — APPROVED** [review]

**Review by GilStein1 — APPROVED** [review]
Very good👌

**Review by maya1414 — APPROVED** [review]

**Review by greenblitz4590 — APPROVED** [review]

**Pull request merged**